d2/src/main/java/com/linkedin/d2/balancer/LoadBalancerServer.java

```diff
@@ -71,4 +71,21 @@ void addUriSpecificProperty(String clusterName,
   void shutdown(Callback<None> callback);
 
   String getConnectString();
+
+  /**
+   * Get announce mode of the server. Some server may have different announce mode, e.g. dual write mode, force announce
+   * mode.
+   */
+  AnnounceMode getAnnounceMode();
+
+  enum AnnounceMode
+  {
+    STATIC_OLD_SR_ONLY,                // statically only announce to old service registry
+    DYNAMIC_OLD_SR_ONLY,               // dynamically only announce to old service registry
+    DYNAMIC_DUAL_WRITE,                // dynamically announce to both service registries
+    DYNAMIC_NEW_SR_ONLY,               // dynamically only announce to new service registry
+    DYNAMIC_FORCE_DUAL_WRITE,          // Using dynamic server yet forced to announce to both service registries
```

Comment:
Could I learn more for this specific scenario?

Reply:
DYNAMIC means using IndisZooKeeperServer, as only that class can listen lix and switch dynamically. So DYNAMIC_FORCE_DUAL_WRITE means using IndisZooKeeperServer but with force announcement mode forcing it to do dual write, please see changes in IndisZooKeeperServer in container PR